tests/functional/s3/test_sync_command.py

```diff
@@ -164,3 +165,17 @@ def test_sync_with_delete_on_downloads(self):
         self.assertEqual(self.operations_called[0][0].name, 'ListObjects')
 
         self.assertFalse(os.path.exists(full_path))
+
+    def test_sync_skips_over_files_deleted_between_listing_and_transfer_attempt(self):
+        full_path = self.files.create_file('foo.txt', 'mycontent')
+        cmdline = '%s %s s3://bucket/' % (
+            self.prefix, self.files.rootdir)
+
+        # FileGenerator.list_files should skip over files that cause an OSError to be raised
+        # because they are missing when we try to get their stats.
+        with patch('awscli.customizations.s3.filegenerator.get_file_stat', side_effect=OSError()):
+            self.run_cmd(cmdline, expected_rc=0)
```

Comment:
I don't think you need to test this in your current setup. I was just saying that even with this fix I do not think it will prevent all nonzero error codes from files being deleted mid-sync and the test as it sits now does not accurately reflect what will actually happen given the current state of the code.

Looking at the code again the issue is not that ``self.parsed_responses`` is not set. The issue is that when [trigger_warnings](https://github.com/aws/aws-cli/pull/2333/files#diff-2d6153b2f8310981a59f57a017f7a2a3R279) is called because of the ``OSError``, the file still exists and therefore fails to meet the check in ``trigger_warning`` for if the file exists.

As to https://github.com/aws/aws-cli/issues/1841 if this merged, that particular issue should be closed as it is only related to the OSError that crashes the ``sync`` command during the stating of the files. The general issue of suppressing errors and warnings for files that are getting modified/deleted mid-sync, that is a much larger problem, a lot harder to solve, and deserves its own conversation/issue.